cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/spark/SparkUtils.java

```diff
@@ -45,16 +47,38 @@
   private static final String SPARK_ASSEMBLY_JAR = "SPARK_ASSEMBLY_JAR";
   // Environment variable name for locating spark home directory
   private static final String SPARK_HOME = "SPARK_HOME";
+  // Environment variable name for locating hadoop and other spark dependencies for hadoop less spark
+  private static final String SPARK_DIST_CLASSPATH = "SPARK_DIST_CLASSPATH";
 
   private static File sparkAssemblyJar;
 
+
+  /**
+   * Locate the spark assembly and its hadoop dependency jars (if hadoop less build) from the local filesystem.
+   *
+   * @return jar location of spark assembly and hadoop dependencies (if hadoop less build)
+   */
+  public static synchronized Set<File> locateSparkDependencyJars() {
```

Comment:
Right. So will it better if we just memorize the Set<File> ? But that assumes that SPARK_DIST_CLASSPATH will remain same all the time whereas its possible that an user might add more jars to it right ?